### PR TITLE
[zh-cn]Sync custom-resources.md

### PR DESCRIPTION
**Pull request by yanrongshi**
Sync Upstream and update format
```yaml
content/zh-cn/docs/concepts/extend-kubernetes/api-extension/custom-resources.md
```
Preview:https://deploy-preview-36347--kubernetes-io-main-staging.netlify.app/zh-cn/docs/concepts/extend-kubernetes/api-extension/custom-resources

### CODE REVIEW

File: content/zh-cn/docs/concepts/extend-kubernetes/api-extension/custom-resources.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ weight: 10
 resource to your Kubernetes cluster and when to use a standalone service. It describes the two
 methods for adding custom resources and how to choose between them.
 -->
-*定制资源（Custom Resource）* 是对 Kubernetes API 的扩展。
+**定制资源（Custom Resource）**是对 Kubernetes API 的扩展。
```
Comment (windsonsea) [inline content/zh-cn/docs/concepts/extend-kubernetes/api-extension/custom-resources.md:22]:
```suggestion
**定制资源（Custom Resource）** 是对 Kubernetes API 的扩展。
```

File: content/zh-cn/docs/concepts/extend-kubernetes/api-extension/custom-resources.md [hunk 2]
```diff
@@ -54,15 +53,14 @@ Once a custom resource is installed, users can create and access its objects usi
 [kubectl](/docs/reference/kubectl/), just as they do for built-in resources like
 *Pods*.
 -->
-*定制资源（Custom Resource）* 是对 Kubernetes API 的扩展，不一定在默认的
-Kubernetes 安装中就可用。定制资源所代表的是对特定 Kubernetes 安装的一种定制。
-不过，很多 Kubernetes 核心功能现在都用定制资源来实现，这使得 Kubernetes
-更加模块化。
+**定制资源（Custom Resource）**是对 Kubernetes API 的扩展，不一定在默认的
```
Comment (windsonsea) [inline content/zh-cn/docs/concepts/extend-kubernetes/api-extension/custom-resources.md:56]:
```suggestion
**定制资源（Custom Resource）** 是对 Kubernetes API 的扩展，不一定在默认的
```

File: content/zh-cn/docs/concepts/extend-kubernetes/api-extension/custom-resources.md [hunk 3]
```diff
@@ -54,15 +53,14 @@ Once a custom resource is installed, users can create and access its objects usi
...
+Kubernetes 安装中就可用。
+定制资源所代表的是对特定 Kubernetes 安装的一种定制。
+不过，很多 Kubernetes 核心功能现在都用定制资源来实现，这使得 Kubernetes更加模块化。
```
Comment (windsonsea) [inline content/zh-cn/docs/concepts/extend-kubernetes/api-extension/custom-resources.md:59]:
```suggestion
不过，很多 Kubernetes 核心功能现在都用定制资源来实现，这使得 Kubernetes 更加模块化。
```

File: content/zh-cn/docs/concepts/extend-kubernetes/api-extension/custom-resources.md [hunk 4]
```diff
@@ -54,15 +53,14 @@ Once a custom resource is installed, users can create and access its objects usi
...
 
-定制资源可以通过动态注册的方式在运行中的集群内或出现或消失，集群管理员可以独立于集群
-更新定制资源。一旦某定制资源被安装，用户可以使用 
-[kubectl](/docs/reference/kubectl/)
-来创建和访问其中的对象，就像他们为 *pods* 这种内置资源所做的一样。
+定制资源可以通过动态注册的方式在运行中的集群内或出现或消失，集群管理员可以独立于集群更新定制资源。
+一旦某定制资源被安装，用户可以使用 [kubectl](/zh-cn/docs/reference/kubectl/)来创建和访问其中的对象，
```
Comment (windsonsea) [inline content/zh-cn/docs/concepts/extend-kubernetes/api-extension/custom-resources.md:62]:
```suggestion
一旦某定制资源被安装，用户可以使用 [kubectl](/zh-cn/docs/reference/kubectl/) 来创建和访问其中的对象，
```

File: content/zh-cn/docs/concepts/extend-kubernetes/api-extension/custom-resources.md [hunk 5]
```diff
@@ -74,20 +72,19 @@ provide a true _declarative API_.
 ## 定制控制器   {#custom-controllers}
 
 就定制资源本身而言，它只能用来存取结构化的数据。
-当你将定制资源与 *定制控制器（Custom Controller）* 相结合时，定制资源就能够
-提供真正的 _声明式 API（Declarative API）_。
+当你将定制资源与**定制控制器（Custom Controller）**结合时，定制资源就能够
+提供真正的**声明式 API（Declarative API）**。
```
Comment (windsonsea) [inline content/zh-cn/docs/concepts/extend-kubernetes/api-extension/custom-resources.md:76]:
```suggestion
当你将定制资源与 **定制控制器（Custom Controller）** 结合时，
定制资源就能够提供真正的 **声明式 API（Declarative API）**。
```

**Review by windsonsea — COMMENTED** [review]
The preview link in your desc section is wrong. The correct links shall be:
https://deploy-preview-36353--kubernetes-io-main-staging.netlify.app/zh-cn/docs/concepts/extend-kubernetes/api-extension/custom-resources/

File: content/zh-cn/docs/concepts/extend-kubernetes/api-extension/custom-resources.md [hunk 6]
```diff
@@ -74,20 +72,19 @@ provide a true _declarative API_.
...
+enforces a separation of responsibilities. You declare the desired state of
+your resource. The Kubernetes controller keeps the current state of Kubernetes
+objects in sync with your declared desired state. This is in contrast to an
+imperative API, where you *instruct* a server what to do.
 -->
-使用[声明式 API](/zh-cn/docs/concepts/overview/kubernetes-api/)，
-你可以 _声明_ 或者设定你的资源的期望状态，并尝试让 Kubernetes 对象的当前状态
-同步到其期望状态。控制器负责将结构化的数据解释为用户所期望状态的记录，并
-持续地维护该状态。
+Kubernetes [声明式 API](/zh-cn/docs/concepts/overview/kubernetes-api/) 强制执行职责分离。
```
Comment (windsonsea) [inline content/zh-cn/docs/concepts/extend-kubernetes/api-extension/custom-resources.md:85]:
```suggestion
Kubernetes [声明式 API](/zh-cn/docs/concepts/overview/kubernetes-api/) 强制对职权做了一次分离操作。
```

File: content/zh-cn/docs/concepts/extend-kubernetes/api-extension/custom-resources.md [hunk 7]
```diff
@@ -74,20 +72,19 @@ provide a true _declarative API_.
...
+你可以声明资源的期望状态，Kubernetes 控制器使 Kubernetes 对象的当前状态与你声明的期望状态保持同步。
```
Comment (windsonsea) [inline content/zh-cn/docs/concepts/extend-kubernetes/api-extension/custom-resources.md:86]:
```suggestion
你声明所用资源的期望状态，而 Kubernetes 控制器使 Kubernetes 对象的当前状态与你所声明的期望状态保持同步。
```

File: content/zh-cn/docs/concepts/extend-kubernetes/api-extension/custom-resources.md (hunk 1 above)
Comment (yanrongshi) [reply content/zh-cn/docs/concepts/extend-kubernetes/api-extension/custom-resources.md:22]: 加粗的中文也需要空格对吗？大佬

File: content/zh-cn/docs/concepts/extend-kubernetes/api-extension/custom-resources.md [hunk 8]
```diff
@@ -74,20 +72,19 @@ provide a true _declarative API_.
 ## 定制控制器   {#custom-controllers}
 
 就定制资源本身而言，它只能用来存取结构化的数据。
-当你将定制资源与 *定制控制器（Custom Controller）* 相结合时，定制资源就能够
-提供真正的 _声明式 API（Declarative API）_。
+当你将定制资源与**定制控制器（Custom Controller）** 结合时，定制资源就能够
+提供真正的**声明式 API（Declarative API）**。
```
Comment (windsonsea) [inline content/zh-cn/docs/concepts/extend-kubernetes/api-extension/custom-resources.md:76]:
```suggestion
当你将定制资源与**定制控制器（Custom Controller）** 结合时，
定制资源就能够提供真正的**声明式 API（Declarative API）**。
```

**Review by Sea-n — COMMENTED** [review]
/approve

**Pull request merged**